zerver/tests/test_upload.py

```diff
@@ -441,6 +441,26 @@ def test_invalid_avatars(self):
 
             self.assert_json_error(result, "Could not decode avatar image; did you upload an image file?")
 
+    def test_delete_avatar(self):
+        # type: () -> None
+        """
+        A DELETE request to /json/users/me/avatar should delete the user avatar and return gravatar URL
+        """
+        self.login("hamlet@zulip.com")
+        hamlet = get_user_profile_by_email("hamlet@zulip.com")
+        hamlet.avatar_source = UserProfile.AVATAR_FROM_USER
+        hamlet.save()
+
+        result = self.client_delete("/json/users/me/avatar")
+        self.assert_json_success(result)
+
+        user_profile = get_user_profile_by_email('hamlet@zulip.com')
+        self.assertEqual(user_profile.avatar_source, UserProfile.AVATAR_FROM_GRAVATAR)
+
+        json = ujson.loads(result.content)
```

Comment:
minor style thing: maybe check the JSON stuff first, then the DB stuff?  I know it's kind of arbitrary, but that's what most of our tests do.